os/hal/ports/STM32/STM32F3xx/stm32_registry.h

```diff
@@ -1307,7 +1307,7 @@
 #define STM32_HAS_ETH                       FALSE
 
 /* EXTI attributes.*/
-#define STM32_EXTI_NUM_LINES                34
+#define STM32_EXTI_NUM_LINES                33

```

Comment:
Full disclosure, I'm not an expert here. I took a quick look at the [reference manual](https://www.st.com/resource/en/reference_manual/dm00094349-stm32f302xbcde-and-stm32f302x68-advanced-armbased-32bit-mcus-stmicroelectronics.pdf) and am seeing 35 lines for STM32F302xB/C/D/E on page 222. Honestly trust your judgment over my own on this; any chance you could briefly explain where we got 33? Is that what was used for 302xE?

Reply:
Oh geez.  I didn't realize that.

Here's how I came to this conclusion:

- Without this change, the ChibiOS tries to assign EXTI33 to COMP7_IRQn.  This is problematic for 2 reasons:
   - The STM32F302xB only has 4 comparators (COMP1, COMP2, COMP4, and COMP6)
   - There are 35 EXTI lines (as @afbowen has pointed out) and ChibiOS maps EXTI33 incorrectly to COMP7 (which should be reserved).

The compilation error we get is:
```
Compiling hal_ext_lld_isr.c
framework/ChibiOS/os/hal/ports/STM32/STM32F3xx/hal_ext_lld_isr.c: In function 'ext_lld_exti_irq_enable':
framework/ChibiOS/os/hal/ports/STM32/STM32F3xx/hal_ext_lld_isr.c:425:20: error: 'COMP7_IRQn' undeclared (first use in this function)
   nvicEnableVector(COMP7_IRQn, STM32_EXT_EXTI33_IRQ_PRIORITY);
                    ^~~~~~~~~~
framework/ChibiOS/os/hal/ports/STM32/STM32F3xx/hal_ext_lld_isr.c:425:20: note: each undeclared identifier is reported only once for each function it appears in
framework/ChibiOS/os/hal/ports/STM32/STM32F3xx/hal_ext_lld_isr.c: In function 'ext_lld_exti_irq_disable':
framework/ChibiOS/os/hal/ports/STM32/STM32F3xx/hal_ext_lld_isr.c:453:21: error: 'COMP7_IRQn' undeclared (first use in this function)
   nvicDisableVector(COMP7_IRQn);
                     ^~~~~~~~~~
make: *** [framework/platforms/ARMCMx/rules.mk:223: build/com.matternet.bmu_1.0/obj/hal_ext_lld_isr.o] Error 1
```
This is error is caused by (hal_ext_lld_isr.c):
```
void ext_lld_exti_irq_enable(void) {

  nvicEnableVector(EXTI0_IRQn, STM32_EXT_EXTI0_IRQ_PRIORITY);
  nvicEnableVector(EXTI1_IRQn, STM32_EXT_EXTI1_IRQ_PRIORITY);
  nvicEnableVector(EXTI2_TSC_IRQn, STM32_EXT_EXTI2_IRQ_PRIORITY);
  nvicEnableVector(EXTI3_IRQn, STM32_EXT_EXTI3_IRQ_PRIORITY);
  nvicEnableVector(EXTI4_IRQn, STM32_EXT_EXTI4_IRQ_PRIORITY);
  nvicEnableVector(EXTI9_5_IRQn, STM32_EXT_EXTI5_9_IRQ_PRIORITY);
  nvicEnableVector(EXTI15_10_IRQn, STM32_EXT_EXTI10_15_IRQ_PRIORITY);
  nvicEnableVector(PVD_IRQn, STM32_EXT_EXTI16_IRQ_PRIORITY);
  nvicEnableVector(RTC_Alarm_IRQn, STM32_EXT_EXTI17_IRQ_PRIORITY);
#if STM32_HAS_USB
  nvicEnableVector(USBWakeUp_IRQn, STM32_EXT_EXTI18_IRQ_PRIORITY);
#endif
  nvicEnableVector(TAMP_STAMP_IRQn, STM32_EXT_EXTI19_IRQ_PRIORITY);
  nvicEnableVector(RTC_WKUP_IRQn, STM32_EXT_EXTI20_IRQ_PRIORITY);
  nvicEnableVector(COMP1_2_3_IRQn, STM32_EXT_EXTI21_22_29_IRQ_PRIORITY);
  nvicEnableVector(COMP4_5_6_IRQn, STM32_EXT_EXTI30_32_IRQ_PRIORITY);
#if STM32_EXTI_NUM_LINES >= 34
  nvicEnableVector(COMP7_IRQn, STM32_EXT_EXTI33_IRQ_PRIORITY);
#endif
}
```
- I suspect (though I cannot say for sure that) Mitch ran into this same problem with the STM32F302RE.

So I thought I had 2 options:
1) adopt Mitch's solution
2) comment out the line assigning COMP7_IRQn

I went with option 1 because:
- I didn't know if going with option 2 would affect other builds
- this option would only affect STM32F302RB and STM32F302RC
- Mitch had already adopted this solution so I assumed (maybe wrongfully) it's safe.

@afbowen Thoughts?

Reply:
That definitely clears it up for me. Thanks for taking the time to explain so thoroughly!

Reply:
Looking at other MCUs in the STM32F3 family, the F358xx and F398xx have a COMP7 which is mapped to EXTI33.  So I think this is an oversight on the ChibiOS team.  Since hal_ext_lld_isr.c, is used by all STM32 compilations, I think our solution is the safest.  This is still a bug in ChibiOS' implementation (IMHO).  Looking at the ChibiOS/ChibiOS repository, the problem is still there.